app/controllers/users_controller.rb

```diff
@@ -0,0 +1,51 @@
+class UsersController < ApplicationController
+  get '/signup' do
+    if !logged_in?
+      erb :'users/create_user'
+    else
+      redirect '/tweets'
+    end
+  end
+
+  post '/signup' do
+    user = User.new(username: params[:username], email: params[:email], password: params[:password])
+    if !user.username.empty? && !user.email.empty? && user.save
```

Comment:
Hey @mendelB, that Learn lesson you linked doesn't actually seem to mention anything about Active Record validations. It says:
> We'll validate some of the attributes of our user by writing code that makes sure no one can sign up without inputing their name, email and password. More on this later.

and then later adds:
> You'll see that you're passing a number of tests, including tests like these:
```ruby
User
is invalid without a name
is invalid without a email
is invalid without an password
```
Let's think about the concept of validations...

but it never actually seems to elaborate on validations, Active Record or otherwise.